### ISSUE TITLE
Need better error message for "IOError: No such file or directory" in environment.py:1272

### ISSUE THREAD

**Issue by ericdand** [sphinx-doc/sphinx#1812]
Hey, I was reconfiguring my project and copied an old conf.py over my new one and an exception occurred: 

```
  File "/usr/local/lib/python2.7/dist-packages/sphinx/environment.py", line 1272, in get_doctree
    f = open(doctree_filename, 'rb')
IOError: [Errno 2] No such file or directory: u'/home/eric/sw/api/core-python/trunk/docs/build/doctrees/index.doctree'
```

It's certainly my own fault, but the message said "Please also report this if it was a user error," so here I am reporting it. :)

Here's the full output of `/tmp/sphinx-err-Zf3N85.log`:

```
# Sphinx version: 1.3.1
# Python version: 2.7.3 (CPython)
# Docutils version: 0.12 release
# Jinja2 version: 2.7.3
# Last messages:
#   looking for now-outdated files...
#   none found
#   pickling environment...
#   done
#   checking consistency...
#   /home/eric/sw/api/core-python/trunk/docs/source/newindex.rst:: WARNING: document isn't included in any toctree
#   done
#   preparing documents...
#   done
#   writing output... [ 50%] index
# Loaded extensions:
#   sphinx.ext.autodoc (1.3.1) from /usr/local/lib/python2.7/dist-packages/sphinx/ext/autodoc.pyc
#   sphinx.ext.intersphinx (1.3.1) from /usr/local/lib/python2.7/dist-packages/sphinx/ext/intersphinx.pyc
#   sphinx.ext.doctest (1.3.1) from /usr/local/lib/python2.7/dist-packages/sphinx/ext/doctest.pyc
#   alabaster (0.7.2) from /usr/local/lib/python2.7/dist-packages/alabaster/__init__.pyc
#   sphinx.ext.coverage (1.3.1) from /usr/local/lib/python2.7/dist-packages/sphinx/ext/coverage.pyc
#   sphinx.ext.pngmath (1.3.1) from /usr/local/lib/python2.7/dist-packages/sphinx/ext/pngmath.pyc
#   sphinx.ext.napoleon (1.3.1) from /usr/local/lib/python2.7/dist-packages/sphinx/ext/napoleon/__init__.pyc
#   sphinx.ext.todo (1.3.1) from /usr/local/lib/python2.7/dist-packages/sphinx/ext/todo.pyc
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/sphinx/cmdline.py", line 245, in main
    app.build(opts.force_all, filenames)
  File "/usr/local/lib/python2.7/dist-packages/sphinx/application.py", line 264, in build
    self.builder.build_update()
  File "/usr/local/lib/python2.7/dist-packages/sphinx/builders/__init__.py", line 245, in build_update
    'out of date' % len(to_build))
  File "/usr/local/lib/python2.7/dist-packages/sphinx/builders/__init__.py", line 316, in build
    self.write(docnames, list(updated_docnames), method)
  File "/usr/local/lib/python2.7/dist-packages/sphinx/builders/__init__.py", line 354, in write
    self._write_serial(sorted(docnames), warnings)
  File "/usr/local/lib/python2.7/dist-packages/sphinx/builders/__init__.py", line 360, in _write_serial
    doctree = self.env.get_and_resolve_doctree(docname, self)
  File "/usr/local/lib/python2.7/dist-packages/sphinx/environment.py", line 1288, in get_and_resolve_doctree
    doctree = self.get_doctree(docname)
  File "/usr/local/lib/python2.7/dist-packages/sphinx/environment.py", line 1272, in get_doctree
    f = open(doctree_filename, 'rb')
IOError: [Errno 2] No such file or directory: u'/home/eric/sw/api/core-python/trunk/docs/build/doctrees/index.doctree'
```

**Comment by shimizukawa** [comment]
Agreed. "better error message" is needed.

**Comment by tk0miya** [comment]
Sphinx already says following warning:
```
WARNING: master file /Users/tkomiya/work/tmp/foo/index.rst not found
```

@shimizukawa I think it is better to replace this warning to SphinxError. What do you think?

**Comment by tk0miya** [comment]
It seems the error message is similar to #2197.
It might fixed by 7bb685b.

**Comment by shimizukawa** [comment]
Definitely right.
Now I close this issue. If the problem is still happened, please feel free to resume at any time. Thanks!
